Source/Scene/QuadtreePrimitive.js

```diff
@@ -109,6 +117,12 @@ define([
         this._lastTileIndex = 0;
         this._updateHeightsTimeSlice = 2.0;
 
+        // If a culled tile contains a cartographic positions in this list, it will be marked
+        // TileSelectionResult.CULLED_BUT_NEEDED and added to the list of tiles to update heights,
+        // even though it is not rendered. The first position will be the position of the
+        // camera and the second will be the origin of the camera's reference frame.
+        this._neededPositions = [undefined, undefined];
+
```

Comment:
Why is `_neededPositions` an array instead of an object with `_neededPositions.camera` and `_neededPositions.cameraReferenceOrigin` ? I think as an array it's more extendable and I assume it can be used for more positions in the future. It just seems potentially error-prone to be making this assumption about what the first and second positions will always be.

Reply:
Yeah, exactly, the idea is that other positions can be added and get the same treatment. But the first two are "reserved" for a specific purpose. Could be over engineering.